### ISSUE TITLE
Listview Scroll jittering when the list item include doc references to other collections. 

### ISSUE THREAD

**Issue by chunfengzhao61** [FlutterFlow/flutterflow-issues#789]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

With a list view vertical or horizontal, if each list item include doc references to other collections, then when scoll the list back slowly, it will be jittering a lot, it will jumpimg back a couple of item when scroll close to list item widget boundary. 
Foward scroll seems fine.  

### Expected Behavior

scroll should be smooth without any jittering or jumping. 

### Steps to Reproduce

1. create a firebase list, each list item should include at least one doc reference to a collection, the collection could be as simple as only to include a single text field.  make list items to up to 20 items. 
2. create a new page
3. add listview widget
4. bind the firebase list to listview widget
5. add container into listview
6. bind the doc reference of list item to the container 
7. at run time, scroll the list all the way to the end first, then slowlly scroll back, it will be jittering when scroll close to a widget boundary, basically jumping forward a couple items, instead of scrolling smoothly.  

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFTzMrqx5V1oddc+KrfLPlGljg6I0MdXZIRmOtvewgjF774PYk2P8+lSFZpTdCzaH9AAVqbpEMKwdb7kt7tFPQCNQWaQYc4/rpuUhPPbGWtaY/RCbW8XGhPN55bN2GM356GkwpRIfthWGRn23GIfuiUQT6Cf9qOYwx5e6fDbOY=

### Context

This bug will make user experience very bad when using my app. 
My app use lot of listviews and it will use doc references to access shared collections for each list item, without it my app will be almost useless. I was trying to work around it, but it just not possible to make it work properly. 

### Visual documentation

https://user-images.githubusercontent.com/116539335/233110590-1a4ae5b4-b0ed-4927-9dab-33d2464e77b3.mp4

### Environment

```markdown
- FlutterFlow version: v3.1 -fllutter v3.7.3
- Platform:  web
- Browser name and version: Chrome 112.0.5615.121
- Operating system and version affected: windows, Android, web
```

**Comment by aemelyanovff** [comment]
Hi @chunfengzhao61.

It happens because, while you are scrolling up, it re-runs queries for items that appear on the screen. Try this:
* Wrap your list items in containers with a fixed height.
* Configure [query caching](https://docs.flutterflow.io/data-and-backend/backend-query#backend-query-caching) for the inner query.

Other than this, I don't think there's anything we can do to help here.

Side note: querying Firestore for each item in a long list can be inefficient. If possible, try reorganizing your Firestore schema in a way that you can get all data to render the ListView in one query.

**Comment by mixart** [comment]
> Hi @chunfengzhao61.
> 
> It happens because, while you are scrolling up, it re-runs queries for items that appear on the screen. Try this:
> 
> * Wrap your list items in containers with a fixed height.
> * Configure [query caching](https://docs.flutterflow.io/data-and-backend/backend-query#backend-query-caching) for the inner query.
> 
> Other than this, I don't think there's anything we can do to help here.
> 
> Side note: querying Firestore for each item in a long list can be inefficient. If possible, try reorganizing your Firestore schema in a way that you can get all data to render the ListView in one query.

For what it's worth, I came across this issue today and I think there's something here. I have a very simple listview, and each row is referencing another collection - not a complex relational query. I spent the past day trying to isolate if it was layout issues, etc. Followed the suggestions above and removed all possible width/height layout issues and it seems to be jittering even with caching and single-time queries.

- Simply make a list with a collection like animals
- make one field have a document reference to another collect to get something like dog > color

Trips
- My california trip
- My France trip > doc reference to a collection of locations > locationName=string: France
- My Europe trip

On each row add a get document from query to display the location name string on the row.

Scrolling upwards creates pretty bad jitters, without even applying any layout complexity. If I find time, I'll try make a project demo.
